src/proxy/createClassProxy.js

```diff
@@ -309,7 +315,9 @@ function createClassProxy(InitialComponent, proxyKey, options) {
 
   update(InitialComponent)
 
-  const proxy = { get, update }
+  proxy = { get, update }
+
+  proxies.set(InitialComponent, proxy)
```

Comment:
This is actually a fix. Everything else is related to this change - now tests have "reset" proxies sometimes.